src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -299,6 +299,13 @@ ExpectedPrivateKey PrivateKey::Load(const Args &args) {
 	if (args.ssl_engine != "") {
 		return LoadFromHSMEngine(args);
 	}
+	// Try to make sure the key has the right permissions and warn if there was
+	// a change (MEN-7752), but try to load it in any case (it may not even be a
+	// file on the file system).
+	if (common::StartsWith<string>(args.private_key_path, "/")) {
+		path::Permissions(
```

Comment:
Unchecked error return.

Reply:
Well, we don't have a way to react to the error except for logging it, but I was afraid it could produce noise for non-standard private key URIs (because it's not exactly a file system path, right?). If you think the prefix check is good enough for this to log and error, I'm all for that.